### PR TITLE
Observe some -Xlint recommendations

### PR DESCRIPTION
**Pull request by som-snytt**
Not all unused things are removed outright.

### CODE REVIEW

File: src/reflect/scala/reflect/internal/StdNames.scala [hunk 1]
```diff
@@ -351,7 +351,7 @@ trait StdNames {
     val MIRROR_UNTYPED: NameType           = "$m$untyped"
     val REIFY_FREE_PREFIX: NameType        = "free$"
     val REIFY_FREE_THIS_SUFFIX: NameType   = "$this"
-    val REIFY_FREE_VALUE_SUFFIX: NameType  = "$value"
+    val REIFY_FREE_VALUE_SUFFIX: NameType  = "$" + "value"
```
Comment (lrytz) [inline src/reflect/scala/reflect/internal/StdNames.scala]: not sure about those.. people will wonder why the definitions here are not all the same style.

File: src/compiler/scala/reflect/reify/phases/Reshape.scala [hunk 2]
```diff
@@ -235,6 +235,7 @@ trait Reshape {
       New(TypeTree(ann.atp) setOriginal extractOriginal(ann.original), List(args))
     }
 
+    /*
     private def toPreTyperLazyVal(ddef: DefDef): ValDef = {
```
Comment (lrytz) [inline src/compiler/scala/reflect/reify/phases/Reshape.scala]: let's remove this, and unused code in general in this PR.

File: src/compiler/scala/tools/cmd/Interpolation.scala [hunk 3]
```diff
@@ -33,15 +33,15 @@ trait Interpolation {
 object Interpolation {
   /** A simple template for generating bash completion functions.
    */
-  lazy val bashTemplate = """
+  lazy val bashTemplate = s"""
```
Comment (lrytz) [inline src/compiler/scala/tools/cmd/Interpolation.scala:36]: i'm not a big fan of this, but OK if you insist :)

File: src/library/scala/Predef.scala [hunk 4]
```diff
@@ -616,7 +616,7 @@ object Predef extends LowPriorityImplicits {
   /** @group type-constraints */
   object <:< {
     /** If `A <: B` and `B <: A`, then `A = B` (subtyping is antisymmetric) */
-    def antisymm[A, B](implicit l: A <:< B, r: B <:< A): A =:= B = =:=.singleton.asInstanceOf[A =:= B]
+    def antisymm[A, B](implicit @deprecated("unused","shim") l: A <:< B, @deprecated("unused","shim") r: B <:< A): A =:= B = =:=.singleton.asInstanceOf[A =:= B]
```
Comment (lrytz) [inline src/library/scala/Predef.scala]: I'd avoid doing this, I find looking at this much more annoying than the lint warning (which we never see). Especially in the standard library, but even in the compiler.

File: src/library/scala/collection/View.scala [hunk 5]
```diff
@@ -143,7 +143,7 @@ object View extends IterableFactory[View] {
   }
 
   /** A view that partitions an underlying collection into two views */
-  @SerialVersionUID(3L)
+  //@SerialVersionUID(3L)
```
Comment (lrytz) [inline src/library/scala/collection/View.scala]: I think the bug is the other way here, so we could leave this for now. https://github.com/scala/bug/issues/10945

File: src/library/scala/collection/immutable/ChampHashMap.scala [hunk 6]
```diff
@@ -182,6 +183,7 @@ private class BitmapIndexedMapNode[K, +V](val dataMap: Int, val nodeMap: Int, va
 
     predicate1 && predicate2 && predicate3
   }
+  */
```
Comment (lrytz) [inline src/library/scala/collection/immutable/ChampHashMap.scala:186]: 👍, this makes sense to keep in comments for a while.

**Review by lrytz — CHANGES_REQUESTED** [review]
I think we can remove unused code in general, the only exception I spotted is the check methods in ChampHashSet/Map, these probably make sense to keep in comments.

File: src/compiler/scala/tools/nsc/io/Socket.scala [hunk 7]
```diff
@@ -37,7 +37,7 @@ class Socket(jsocket: JSocket) extends Streamable.Bytes with Closeable {
   def close()        = jsocket.close()
 
   def printWriter()                         = new PrintWriter(outputStream(), true)
-  def bufferedReader(implicit codec: Codec) = new BufferedReader(new InputStreamReader(inputStream()))
+  def bufferedReader(implicit codec: Codec) = new BufferedReader(new InputStreamReader(inputStream(), codec.decoder))
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/io/Socket.scala:40]: A potential bug!

File: build.sbt [hunk 8]
```diff
@@ -158,6 +158,8 @@ lazy val commonSettings = instanceSettings ++ clearSourceAndResourceDirectories
     // END: Copy/pasted from SBT
   },
   fork in run := true,
+  //scalacOptions += "-Xlint:-nullary-override,-by-name-right-associative,-inaccessible,_",
+  //scalacOptions ++= Seq("-Xmaxerrs", "5", "-Xmaxwarns", "5"),
```
Comment (som-snytt) [inline build.sbt:162]: placeholders.

File: src/compiler/scala/reflect/reify/phases/Reshape.scala [hunk 9]
```diff
@@ -248,6 +249,7 @@ trait Reshape {
       val mods2 = toPreTyperModifiers(mods1, ddef.symbol)
       ValDef(mods2, name1, tpt0, extractRhs(rhs0))
     }
+    */
```
Comment (som-snytt) [inline src/compiler/scala/reflect/reify/phases/Reshape.scala]: First, do no harm.

File: src/compiler/scala/tools/nsc/symtab/SymbolTrackers.scala [hunk 10]
```diff
@@ -182,9 +182,7 @@ trait SymbolTrackers {
             val xs = if (back.isEmpty) front else front :+ "..."
             xs mkString " -> "
         }
-        val treeStrings = symMap(sym) map { t =>
-          "%10s: %s".format(t.shortClass, t)
-        }
+        val treeStrings = symMap(sym).map(t => s"${t.shortClass}%10s: $t")
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/symtab/SymbolTrackers.scala:185]: For a moment, I forgot the s interpolator. Luckily, -Xlint told me.

Actually, -Xlint complained about `:+` on L182. Still need to follow up.

File: src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala [hunk 11]
```diff
@@ -1209,7 +1208,7 @@ abstract class ClassfileParser {
       assert(runtimeAnnotStart != -1, s"No RuntimeVisibleAnnotations in classfile with ScalaSignature attribute: $clazz")
       in.bp = runtimeAnnotStart
       val numAnnots = u2
-      var i = 0
+      val i = 0
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala:1211]: Not obvious whether it's a cut/paste bug.

File: src/compiler/scala/tools/nsc/transform/patmat/PatternMatching.scala [hunk 12]
```diff
@@ -65,7 +65,7 @@ trait PatternMatching extends Transform
           // Keep 2.12 behaviour of using wildcard expected type, recomputing the LUB, then throwing it away for the continuations plugins
           // but for the rest of us pass in top as the expected type to avoid waste.
           val pt = if (origTp <:< definitions.AnyTpe) definitions.AnyTpe else WildcardType
-          localTyper.typed(translated, definitions.AnyTpe) setType origTp
+          localTyper.typed(translated, pt) setType origTp
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/transform/patmat/PatternMatching.scala:68]: `-Xlint` would have told you that. -- Paul

File: src/compiler/scala/tools/nsc/typechecker/PatternTypers.scala [hunk 13]
```diff
@@ -44,7 +44,7 @@ trait PatternTypers {
     import TyperErrorGen._
     import infer._
 
-    private def unit = context.unit
+    //private def unit = context.unit
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/typechecker/PatternTypers.scala]: can probably zap aliases.

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 14]
```diff
@@ -5116,7 +5115,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         }
       }
 
-      def typedTypeSelectionQualifier(tree: Tree, pt: Type = AnyRefTpe) =
+      def typedTypeSelectionQualifier(tree: Tree, pt: Type /*= AnyRefTpe*/) =
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: documentary default, removing unused default.

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 15]
```diff
@@ -5557,7 +5556,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       }
 
       def typedSingletonTypeTree(tree: SingletonTypeTree) = {
-        val refTyped = typedTypeSelectionQualifier(tree.ref, WildcardType )
+        val refTyped = typedTypeSelectionQualifier(tree.ref, WildcardType)
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:5562]: collateral whitespace change!

File: src/library/scala/Predef.scala (hunk 4 above)
Comment (som-snytt) [inline src/library/scala/Predef.scala]: Made a PR to ignore these cases.

File: src/library/scala/collection/ArrayOps.scala [hunk 16]
```diff
@@ -1129,7 +1129,7 @@ final class ArrayOps[A](val xs: Array[A]) extends AnyVal {
     else {
       def mkRowBuilder() = ArrayBuilder.make[B](ClassTag[B](aClass.getComponentType))
       val bs = new ArrayOps(asArray(xs(0))).map((x: B) => mkRowBuilder())
-      var j = 0
+      //var j = 0
```
Comment (som-snytt) [inline src/library/scala/collection/ArrayOps.scala]: safe to del.

File: src/library/scala/collection/View.scala (hunk 5 above)
Comment (som-snytt) [inline src/library/scala/collection/View.scala]: These warnings in the build should be resolved. Also, maybe the warning should sit behind a flag. `-Xlint:serial`? `-Ywarn-serial`? `-Ywarn-srsly`?

File: src/library/scala/concurrent/Channel.scala [hunk 17]
```diff
@@ -17,11 +17,11 @@ package scala.concurrent
  *  @author  Martin Odersky
  */
 class Channel[A] {
-  class LinkedList[A] {
+  private class LinkedList {
```
Comment (som-snytt) [inline src/library/scala/concurrent/Channel.scala:20]: we're all friends here.

File: src/reflect/scala/reflect/api/Trees.scala [hunk 18]
```diff
@@ -2165,7 +2165,8 @@ trait Trees { self: Universe =>
   /** 0-1 argument list new, based on a symbol.
    *  @group Factories
    */
-  @deprecated("use q\"new ${sym.toType}(..$args)\" instead", "2.10.1")
+  private[this] final val depstr_New = "use q\"new ${sym." + "toType}(..$args)\" instead"
+  @deprecated(depstr_New, "2.10.1")
```
Comment (som-snytt) [inline src/reflect/scala/reflect/api/Trees.scala]: implicitNotFound gets a pass, but actually all ordinary annotations? I think macros supplying constants would be OK.

File: src/reflect/scala/reflect/api/TypeTags.scala [hunk 19]
```diff
@@ -300,6 +300,7 @@ trait TypeTags { self: Universe =>
   /* @group TypeTags */
   // This class only exists to silence MIMA complaining about a binary incompatibility.
   // Only the top-level class (api.PredefTypeCreator) should be used.
+  @deprecated("This class only exists to silence MIMA complaining about a binary incompatibility.", since="forever")
```
Comment (som-snytt) [inline src/reflect/scala/reflect/api/TypeTags.scala:303]: didn't work

File: src/reflect/scala/reflect/internal/Types.scala [hunk 20]
```diff
@@ -2501,6 +2501,7 @@ trait Types
         super.prefixString
     )
     // Suppressing case class copy method which risks subverting our single point of creation.
+    @deprecated("Suppressing case class copy method", since="forever")
```
Comment (som-snytt) [inline src/reflect/scala/reflect/internal/Types.scala:2497]: also didn't work.

File: src/reflect/scala/reflect/internal/transform/UnCurry.scala [hunk 21]
```diff
@@ -108,7 +108,7 @@ trait UnCurry {
     // we are using `origSym.info`, which contains the type *before* the transformation
     // so we still see repeated parameter types (uncurry replaces them with Seq)
     val isRepeated = origSym.info.paramss.flatten.map(sym => definitions.isRepeatedParamType(sym.tpe))
-    val oldPs = newInfo.paramss.head
+    val _ = newInfo.paramss.head
```
Comment (som-snytt) [inline src/reflect/scala/reflect/internal/transform/UnCurry.scala]: wary of side effects, left it in.

File: src/reflect/scala/reflect/io/IOStats.scala [hunk 22]
```diff
@@ -1,7 +1,7 @@
 package scala
 package reflect.io
 
-import scala.reflect.internal.util.Statistics
+//import scala.reflect.internal.util.Statistics
```
Comment (som-snytt) [inline src/reflect/scala/reflect/io/IOStats.scala:4]: deletable

File: src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala [hunk 23]
```diff
@@ -944,10 +944,10 @@ class ILoop(config: ShellConfig, inOverride: BufferedReader = null,
   /** Actions to cram in parallel while collecting first user input at prompt.
     * Run with output muted both from ILoop and from the intp reporter.
     */
-  private def interpretPreamble = {
+  private def interpretPreamble() = {
```
Comment (som-snytt) [inline src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala:947]: left most "parens mean side effects" alone.

File: src/reflect/scala/reflect/internal/StdNames.scala (hunk 1 above)
Comment (som-snytt) [reply src/reflect/scala/reflect/internal/StdNames.scala]: I'll check if intrinsic `s` interpolator does a constant.

s/instrinsic/instringsic/

File: src/compiler/scala/reflect/reify/phases/Reshape.scala (hunk 2 above)
Comment (som-snytt) [reply src/compiler/scala/reflect/reify/phases/Reshape.scala]: OK. I'm a hoarder.

File: src/compiler/scala/tools/cmd/Interpolation.scala (hunk 3 above)
Comment (som-snytt) [reply src/compiler/scala/tools/cmd/Interpolation.scala:36]: This was the one I liked! As in, this is canonical.

File: src/library/scala/Predef.scala (hunk 4 above)
Comment (som-snytt) [reply src/library/scala/Predef.scala]: Yes, I'll rebase over the PR which doesn't warn. Because I see the warnings. At night in my dreams.

Comment (som-snytt) [reply src/library/scala/Predef.scala]: Actually, this was at the beginning of the edits, when I thought I wanted to minimize the compiler output, but by the end it wasn't so bad, with deprecations turned off of course.

File: src/library/scala/collection/View.scala (hunk 5 above)
Comment (som-snytt) [reply src/library/scala/collection/View.scala]: As I suspected, thx.

File: src/compiler/scala/tools/cmd/Interpolation.scala (hunk 3 above)
Comment (som-snytt) [reply src/compiler/scala/tools/cmd/Interpolation.scala:36]: Confirmed that intrinsic s just yields a constant string, as desired.

File: src/reflect/scala/reflect/io/IOStats.scala (hunk 22 above)
Comment (som-snytt) [reply src/reflect/scala/reflect/io/IOStats.scala:4]: actually all that file is commented out.

**Comment by som-snytt** [comment]
Feedback is addressed. Hopefully will be merged before it fails to merge.

File: src/reflect/scala/reflect/internal/StdNames.scala [hunk 24]
```diff
@@ -351,7 +351,7 @@ trait StdNames {
     val MIRROR_UNTYPED: NameType           = "$m$untyped"
     val REIFY_FREE_PREFIX: NameType        = "free$"
     val REIFY_FREE_THIS_SUFFIX: NameType   = "$this"
-    val REIFY_FREE_VALUE_SUFFIX: NameType  = "$value"
+    val REIFY_FREE_VALUE_SUFFIX: NameType  = s"$$value"      // looks like missing interpolator due to `value` in scopre
```
Comment (som-snytt) [inline src/reflect/scala/reflect/internal/StdNames.scala]: Unfortunately, `StringContext` name hides the interpolator. Ironic!

File: src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala (hunk 11 above)
Comment (lrytz) [reply src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala:1211]: This is an oversight, `i` should be incremented in the loop. https://github.com/scala/scala/pull/6816

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala (hunk 14 above)
Comment (lrytz) [reply src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Happy to have the default removed

File: src/reflect/scala/reflect/internal/StdNames.scala [hunk 25]
```diff
@@ -351,7 +358,7 @@ trait StdNames {
     val MIRROR_UNTYPED: NameType           = "$m$untyped"
     val REIFY_FREE_PREFIX: NameType        = "free$"
     val REIFY_FREE_THIS_SUFFIX: NameType   = "$this"
```
Comment (lrytz) [inline src/reflect/scala/reflect/internal/StdNames.scala:360]: was there no warning here?

File: src/reflect/scala/reflect/internal/StdNames.scala [hunk 26]
```diff
@@ -312,6 +314,11 @@ trait StdNames {
     override type NameType = TermName
 
     protected implicit def createNameType(name: String): TermName = newTermNameCached(name)
+    // create a name with leading dollar, avoiding the appearance that interpolation of a value (possibly named "value") was intended
+    private class NameContext(s: String) {
+      def n(args: Any*): TermName = { require(args.isEmpty) ; createNameType("$" + s) }
+    }
+    private def StringContext(parts: String*) = { require(parts.size == 1) ; new NameContext(parts.head) }
```
Comment (lrytz) [inline src/reflect/scala/reflect/internal/StdNames.scala:321]: 👍 haven't seen this before

**Review by lrytz — COMMENTED** [review]
> Hopefully will be merged before it fails to merge.

😢

**Comment by lrytz** [comment]
LGTM otherwise, let's merge quickly after the rebase.

File: src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala (hunk 11 above)
Comment (som-snytt) [reply src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala:1211]: Reverted to var.

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala (hunk 14 above)
Comment (som-snytt) [reply src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Made non-documentary.

File: src/reflect/scala/reflect/internal/StdNames.scala (hunk 25 above)
Comment (som-snytt) [reply src/reflect/scala/reflect/internal/StdNames.scala:360]: Sharp eyes! I'll wonder if this isPrivateLocal when I'm fully awake.

File: src/reflect/scala/reflect/internal/StdNames.scala (hunk 26 above)
Comment (som-snytt) [reply src/reflect/scala/reflect/internal/StdNames.scala:321]: The puzzler guy didn't like my one-liner StringContext puzzler. :(  It was too improbable.

**Review by lrytz — APPROVED** [review]

**Comment by som-snytt** [comment]
@lrytz Thanks for the review of tedious material. You make a PR better than it deserves to be!

File: src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala (hunk 11 above)
Comment (som-snytt) [reply src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala:1211]: You said you reverted it. Maybe you inverted it.

**Comment by som-snytt** [comment]
There are no plain strings. Added `f` for emphasis.
```
[error] /Users/andrew/workspace/scala/src/compiler/scala/tools/util/SocketServer.scala:98: conversions must follow a splice; use %% for literal %, %n for newline
[error]           warn(f"Accept on port %d failed")
```
One could lint for format strings. Yesterday, in Java land, I added a private utility for the formatted message style my colleague prefers, `formatted("{} and {}", objects...)`, when I happened to notice that I omitted the `ted` and it still compiled: the file had a static import of `String.format`.

Also the `f-interpolator` could be improved, as it's obviously a missing arg.

File: src/reflect/scala/reflect/internal/StdNames.scala (hunk 25 above)
Comment (som-snytt) [reply src/reflect/scala/reflect/internal/StdNames.scala:360]: I need a couple more days of sleep after the national election, then I'll wonder this. Or `"$this"`.

File: src/reflect/scala/reflect/internal/StdNames.scala (hunk 26 above)
Comment (som-snytt) [reply src/reflect/scala/reflect/internal/StdNames.scala:321]: For some reason, we aren't allowed to have fun things.

**Pull request merged**